### ISSUE TITLE
MembersIdentity doesn't have ClaimsPrincipalFactory

### ISSUE THREAD

**Issue by jbreuer** [umbraco/Umbraco-CMS#12873]
### Which *exact* Umbraco version are you using? For example: 9.0.1 - don't just write v9

10.1

### Bug summary

If you're using an external login provider for members it's not possible to add any extra claims in the `OnExternalLogin` event. This is possible for users because they have a ClaimsPrincipalFactory.

### Specifics

This is in the UmbracoBuilder.BackOfficeIdentity.cs:
```
builder.BuildUmbracoBackOfficeIdentity()
    .AddDefaultTokenProviders()
    .AddUserStore<IUserStore<BackOfficeIdentityUser>, BackOfficeUserStore>(factory => new BackOfficeUserStore(
        factory.GetRequiredService<ICoreScopeProvider>(),
        factory.GetRequiredService<IUserService>(),
        factory.GetRequiredService<IEntityService>(),
        factory.GetRequiredService<IExternalLoginWithKeyService>(),
        factory.GetRequiredService<IOptionsSnapshot<GlobalSettings>>(),
        factory.GetRequiredService<IUmbracoMapper>(),
        factory.GetRequiredService<BackOfficeErrorDescriber>(),
        factory.GetRequiredService<AppCaches>(),
        factory.GetRequiredService<ITwoFactorLoginService>()
    ))
    .AddUserManager<IBackOfficeUserManager, BackOfficeUserManager>()
    .AddSignInManager<IBackOfficeSignInManager, BackOfficeSignInManager>()
    .AddClaimsPrincipalFactory<BackOfficeClaimsPrincipalFactory>()
    .AddErrorDescriber<BackOfficeErrorDescriber>();
```
As you can see claims will be added/transformed in the AddClaimsPrincipalFactory.

This is in the UmbracoBuilder.MembersIdentity.cs:
```
services.AddIdentity<MemberIdentityUser, UmbracoIdentityRole>()
    .AddDefaultTokenProviders()
    .AddUserStore<IUserStore<MemberIdentityUser>, MemberUserStore>(factory => new MemberUserStore(
        factory.GetRequiredService<IMemberService>(),
        factory.GetRequiredService<IUmbracoMapper>(),
        factory.GetRequiredService<ICoreScopeProvider>(),
        factory.GetRequiredService<IdentityErrorDescriber>(),
        factory.GetRequiredService<IPublishedSnapshotAccessor>(),
        factory.GetRequiredService<IExternalLoginWithKeyService>(),
        factory.GetRequiredService<ITwoFactorLoginService>()))
    .AddRoleStore<MemberRoleStore>()
    .AddRoleManager<IMemberRoleManager, MemberRoleManager>()
    .AddMemberManager<IMemberManager, MemberManager>()
    .AddSignInManager<IMemberSignInManager, MemberSignInManager>()
    .AddErrorDescriber<MembersErrorDescriber>()
    .AddUserConfirmation<UmbracoUserConfirmation<MemberIdentityUser>>();
```
As you can see there no extension which adds/transforms the claims.

### Steps to reproduce

Follow the steps to reproduce from this issue: https://github.com/umbraco/Umbraco-CMS/issues/12749

Once you have a project with external login providers for members try to add a claim in the `OnExternalLogin` event. These claims are never added to the member.

### Expected result / actual result

It should be possible to add claims to members the same way as can be done for users.

**Comment by jbreuer** [comment]
@nikolajlauridsen @bergmania @lassefredslund Sorry that I keep spamming you. Since 10.2 already has some fixes for external login providers it would be awesome if this could be fixed as well.

**Comment by nikolajlauridsen** [comment]
Fixed in #12877, thanks heaps for all the amazing work helping us make external logins better 🎉 💪
